features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Create/GraphQL/APICreateGrpahQL.jsx

```diff
@@ -0,0 +1,369 @@
+                }
+                this.setState({ valid: validNew });
+                return;
+            }
+            // No errors so let's fill the inputs with the graphQLBean
+            if (graphQLBean.info) {
+                if (graphQLBean.info.operations) {
+                    this.setState(({ api }) => {
+                        const changes = api;
+                        changes.operations = graphQLBean.info.operations;
+                        changes.gatewayEnvironments = ['Production and Sandbox'];
+                        return { api: changes };
+                    });
+                }
+            }
+            this.setState({
+                activeStep: activeStep + 1,
+            });
+        } else if (activeStep === 1) {
+            // Handling Finish step ( validating the input fields )
+            const { api: currentAPI } = this.state;
+            if (!currentAPI.name || !currentAPI.context || !currentAPI.version || !currentAPI.endpointConfig) {
+                // Checking the api name,version,context undefined or empty states
+                this.setState((oldState) => {
+                    const { valid: isValid, api } = oldState;
+                    const validUpdated = isValid;
+                    validUpdated.name.empty = !api.name;
+                    validUpdated.context.empty = !api.context;
+                    validUpdated.version.empty = !api.version;
+                    validUpdated.endpointConfig.empty = !api.endpointConfig;
+                    return { valid: validUpdated };
+                });
+                return;
+            }
+            this.createGraphQLAPI();
+        }
+    };
+
+    handleBack = () => {
+        this.setState(state => ({
+            activeStep: state.activeStep - 1,
+        }));
+    };
+
+    handleReset = () => {
+        this.setState({
+            activeStep: 0,
+        });
+    };
+
+    /**
+     *
+     *
+     * @returns
+     * @memberof APICreateGraphQL
+     */
+    render() {
+        const { classes } = this.props;
+        const steps = getSteps();
+        const {
+            doValidate, activeStep, graphQLBean, api, valid, loading,
+        } = this.state;
+        const uploadMethod = 'file';
+        const provideGraphQLProps = {
+            [uploadMethod]: graphQLBean[uploadMethod],
+            updateGraphQLBean: this.updateGraphQLBean,
+            validate: doValidate,
+            valid,
+            updateFileErrors: this.updateFileErrors,
+        };
+        if (loading) {
+            return <Progress />;
+        }
+        return (
+            <Grid container spacing={24} className={classes.root}>
+                <Grid item xs={12} xl={6}>
+                    <div className={classes.titleWrapper}>
+                        <Typography variant='h4' align='left' className={classes.mainTitle}>
+                            <FormattedMessage
+                                id='design.a.new.graphQL.api.using.SDL'
```

Comment:
Lets do this as mentioned in [1].

[1] - https://github.com/wso2/carbon-apimgt/wiki/How-internationalization-(i18n)-works-in-API-Manager-React-Apps